crates/astria-sequencer/src/config.rs

```diff
@@ -27,60 +18,26 @@ pub struct Config {
     pub log: String,
 }
 
-impl Config {
-    fn from_environment(envar_prefix: &str) -> Result<Config, figment::Error> {
-        Figment::new()
-            .merge(Env::prefixed("RUST_").split("_").only(&["log"]))
-            .merge(Env::prefixed(envar_prefix))
-            .extract()
-    }
-}
-
 #[cfg(test)]
-mod tests {
-    use figment::Jail;
-    use once_cell::sync::Lazy;
-    use regex::Regex;
+mod test {
+    use astria_utils::{
+        config_test_suite_failing,
+        config_test_suite_passing,
+    };
 
-    use super::Config;
-    const EXAMPLE_ENV: &str = include_str!("../local.env.example");
+    use crate::Config;
 
-    fn populate_environment_from_example(jail: &mut Jail, test_envar_prefix: &str) {
-        static RE_START: Lazy<Regex> = Lazy::new(|| Regex::new(r"^[[:space:]]+").unwrap());
-        static RE_END: Lazy<Regex> = Lazy::new(|| Regex::new(r"[[:space:]]+$").unwrap());
-        for line in EXAMPLE_ENV.lines() {
-            if let Some((key, val)) = line.trim().split_once('=') {
-                assert!(
-                    !(RE_END.is_match(key) || RE_START.is_match(val)),
-                    "env vars must not contain spaces in assignment\n{line}"
-                );
-                let prefixed_key = format!("{test_envar_prefix}_{key}");
-                jail.set_env(prefixed_key, val);
-            }
-        }
-    }
+    const EXAMPLE_ENV: &str = include_str!("../local.env.example");
+    const ENV_PREFIX: &str = "ASTRIA_SEQUENCER_";
```

Comment:
This is pretty boilerplatey - see my comment on the `AstriaConfig` trait using an associated const in the trait that sets this for you.